src/ci/init_repo.sh

```diff
@@ -8,19 +8,17 @@ set -o errexit
 set -o pipefail
 set -o nounset
 
-ci_dir=$(cd $(dirname $0) && pwd)
+ci_dir=$(cd "$(dirname "$0")" && pwd)
 . "$ci_dir/shared.sh"
 
 REPO_DIR="$1"
 CACHE_DIR="$2"
 
-cache_src_dir="$CACHE_DIR/src"
-
-if [ ! -d "$REPO_DIR" -o ! -d "$REPO_DIR/.git" ]; then
+if ! { [ -d "$REPO_DIR" ] && [ ! -d "$REPO_DIR/.git" ]; }; then
```

Comment:
You have an extra `!` in here.
```suggestion
if ! { [ -d "$REPO_DIR" ] && [ -d "$REPO_DIR/.git" ]; }; then
```

Reply:
Ooops, fixed